data/mods/pokebilities/scripts.ts

```diff
@@ -142,5 +142,63 @@ export const Scripts: ModdedBattleScriptsData = {
 
 			return true;
 		},
+		/**
+		 * Changes this Pokemon's forme to match the given speciesId (or species).
+		 * This function handles all changes to stats, ability, type, species, etc.
+		 * as well as sending all relevant messages sent to the client.
+		 */
+		formeChange(speciesId, source, isPermanent?, message?) {
+			if (!source) source = this.battle.effect;
+
+			const rawSpecies = this.battle.dex.species.get(speciesId);
+
+			const species = this.setSpecies(rawSpecies, source);
+			if (!species) return false;
+
+			if (this.battle.gen <= 2) return true;
+
+			// The species the opponent sees
+			const apparentSpecies =
+				this.illusion ? this.illusion.species.name : species.baseSpecies;
+			if (isPermanent) {
+				this.baseSpecies = rawSpecies;
+				this.details = species.name + (this.level === 100 ? '' : ', L' + this.level) +
+					(this.gender === '' ? '' : ', ' + this.gender) + (this.set.shiny ? ', shiny' : '');
+				this.battle.add('detailschange', this, (this.illusion || this).details);
+				if (source.effectType === 'Item') {
+					if (source.zMove) {
+						this.battle.add('-burst', this, apparentSpecies, species.requiredItem);
+						this.moveThisTurnResult = true; // Ultra Burst counts as an action for Truant
+					} else if (source.onPrimal) {
+						if (this.illusion) {
+							this.ability = '';
+							this.battle.add('-primal', this.illusion);
+						} else {
+							this.battle.add('-primal', this);
+						}
+					} else {
+						this.battle.add('-mega', this, apparentSpecies, species.requiredItem);
+						this.moveThisTurnResult = true; // Mega Evolution counts as an action for Truant
+					}
+				} else if (source.effectType === 'Status') {
+					// Shaymin-Sky -> Shaymin
+					this.battle.add('-formechange', this, species.name, message);
+				}
+			} else {
+				if (source.effectType === 'Ability') {
+					this.battle.add('-formechange', this, species.name, message, `[from] ability: ${source.name}`);
+				} else {
+					this.battle.add('-formechange', this, this.illusion ? this.illusion.species.name : species.name, message);
+				}
+			}
+			if (isPermanent && !['disguise', 'iceface', 'ability:disguise', 'ability:iceface'].includes(source.id)) {
```

Comment:
This change, right? Thank-you.

https://github.com/smogon/pokemon-showdown/commit/621d55646a8272205b684452c33d8a63ab35779a

At least for Pokébilities it seems to be OK for Darmanitan because for some reason the source.id sent is 'zenmode' rather than 'ability:zenmode', even when it isn't the real ability. So, if it is safe in general to revert this method, that could work without needing a custom override.